src/cs-core/GuiManager.cpp

```diff
@@ -50,36 +51,33 @@ GuiManager::GuiManager(std::shared_ptr<const Settings> const& settings,
   mViewportUpdater = new VistaViewportResizeToProjectionAdapter(pViewport);
   mViewportUpdater->SetUpdateMode(VistaViewportResizeToProjectionAdapter::MAINTAIN_HORIZONTAL_FOV);
 
-  // Hide the user interface when ESC is pressed.
-  mInputManager->sOnEscapePressed.connect([this]() { toggleGui(); });
-
   // Create GuiAreas and attach them to the SceneGraph ---------------------------------------------
 
   // The global GUI is drawn in world-space, the local GUI is drawn in screen-space.
   auto pSG           = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
   mLocalGuiTransform = pSG->NewTransformNode(pSG->GetRoot());
 
   // The global GUI area is only created when the according settings key was specified.
-  if (settings->mGui) {
+  if (mSettings->mGuiPosition) {
     auto platform = GetVistaSystem()
                         ->GetPlatformFor(GetVistaSystem()->GetDisplayManager()->GetDisplaySystem())
                         ->GetPlatformNode();
     mGlobalGuiTransform = pSG->NewTransformNode(platform);
 
-    mGlobalGuiTransform->Scale(
-        (float)settings->mGui->mWidthMeter, (float)settings->mGui->mHeightMeter, 1.0);
+    mGlobalGuiTransform->Scale((float)mSettings->mGuiPosition->mWidthMeter,
+        (float)mSettings->mGuiPosition->mHeightMeter, 1.0);
     mGlobalGuiTransform->Rotate(
-        VistaAxisAndAngle(VistaVector3D(1, 0, 0), (float)settings->mGui->mRotX));
+        VistaAxisAndAngle(VistaVector3D(1, 0, 0), (float)mSettings->mGuiPosition->mRotX));
```

Comment:
```suggestion
        VistaVector3D(1, 0, 0), static_cast<float>(mSettings->mGuiPosition->mRotX)));
```

Reply:
You're right but in order to avoid too many conflicts with the `feature/clang-tidy` I will do this after it got merged.